### PR TITLE
Refactor attribute processing

### PR DESCRIPTION
**Pull request by stevenbfine**
Refactoring how an element's attributes are processed into a series of helper functions.

### CODE REVIEW

File: content_script.js [hunk 1]
```diff
@@ -14,8 +14,7 @@ class HTMLSerializer {
      *     ignored while serializing a document.
      * @const
      */
-     // TODO(sfine): Process links.
-    this.FILTERED_TAGS = new Set(['script', 'noscript', 'style', 'link']);
+    this.FILTERED_TAGS = new Set(['script', 'noscript', 'style']);
```
Comment (chrishtr) [inline content_script.js]: Is the change on this line part of the refactoring somehow?

File: content_script.js [hunk 2]
```diff
@@ -134,6 +89,91 @@ class HTMLSerializer {
...
+  /**
+   * Process the src attribute of a given element.
+   *
+   * @param {Element} element The element being processed, which has the src
+   *     attribute.
+   * @private
+   */
+  processSrcAttribute(element) {
+    switch(element.tagName.toLowerCase()) {
+      default:
```
Comment (chrishtr) [inline content_script.js]: does the "case 'iframe" condition have any effect? it's after a default:

File: content_script.js [hunk 3]
```diff
@@ -134,6 +89,91 @@ class HTMLSerializer {
   }
 
   /**
+   * Takes an html element, and populates this object's fields with the
+   * appropriate attribute names and values.
+   *
+   * @param {Element} element The Element to serialize.
+   * @private
+   */ 
+  processAttributes(element) {
```
Comment (chrishtr) [inline content_script.js:98]: This is a class function right? Why does it look like it's declared as a static global?

File: content_script.js [hunk 4]
```diff
@@ -134,6 +89,91 @@ class HTMLSerializer {
...
+  }
+
+  /**
+   * Add an entry to |this.srcHoles| so it can be processed asynchronously.
+   *
+   * @param {Element} element The element being processed, which has the src
+   *     attribute.
+   * @private
+   */
+  addSrcHole(element) {
```
Comment (chrishtr) [inline content_script.js]: "add" vs "process" is in consistent among your helper methods.

File: tests/tests.js [hunk 5]
```diff
@@ -76,7 +76,6 @@ QUnit.test('iframeFullyQualifiedName: multiple layers', function(assert) {
   childFrameBody.appendChild(grandChildFrame1);
   childFrameBody.appendChild(grandChildFrame2);
   assert.equal(serializer.iframeFullyQualifiedName(grandChildFrame1.contentWindow), '0.0.0');
-  assert.equal(serializer.iframeFullyQualifiedName(grandChildFrame1.contentWindow), '0.0.0');
```
Comment (chrishtr) [inline tests/tests.js:79]: Why delete this line?

File: content_script.js (hunk 1 above)
Comment (stevenbfine) [inline content_script.js]: No.  In general, for one line/small changes, how should I incorporate them into the pull request process?  Should I:
a) Create a separate pull request.
b) Lump them into another pull request.
c) Just commit them directly to master.

I suppose I just went with option b.  Should I undo it?

File: content_script.js (hunk 2 above)
Comment (stevenbfine) [inline content_script.js]: The purpose of "case 'iframe'" is to simply do nothing.  Should I have put it before the default with a break statement?

File: content_script.js (hunk 4 above)
Comment (stevenbfine) [inline content_script.js]: Done

File: content_script.js (hunk 3 above)
Comment (stevenbfine) [inline content_script.js:98]: I think I'm a little mixed up about the terminology.  I thought that a class method is the same as a static method?  Additionally, I want this method to be an instance method, but I'm not sure what's different about this method than the other methods (most of which I also want to be instance methods).  Is the comment format incorrect?

File: tests/tests.js (hunk 5 above)
Comment (stevenbfine) [inline tests/tests.js:79]: I think I accidentally copied the same line twice, so I deleted the second one.  This is another instance of me lumping a small change into a pull request (as mentioned above).

File: content_script.js (hunk 1 above)
Comment (chrishtr) [inline content_script.js]: Yes, remove it from this pull request. That makes the refactoring review easier.

File: content_script.js (hunk 2 above)
Comment (chrishtr) [inline content_script.js]: Yes.

But if it's just two values, you might as well use if... else.

File: tests/tests.js (hunk 5 above)
Comment (chrishtr) [inline tests/tests.js:79]: This one is fine, since it's just deleting a duplicate.

File: content_script.js (hunk 3 above)
Comment (chrishtr) [inline content_script.js:98]: Per offline conversation, you're right. I had missed the indentation.

File: content_script.js (hunk 1 above)
Comment (stevenbfine) [inline content_script.js]: Done

File: content_script.js (hunk 2 above)
Comment (stevenbfine) [inline content_script.js]: Done

**Comment by chrishtr** [comment]
LGTM

**Pull request merged**